src/components/graph-view.js

```diff
@@ -1127,16 +1126,24 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
     this.setState(newState);
 
     if (!creatingEdge) {
-      // If a group of nodes are selected, then clicking
-      // one of the nodes in the group shouldn't deselect it,
-      // instead the user must click something else.
-      const foundSelectedNode = (selectedNodes || new Set()).has(node);
+      if (allowMultiselect) {
+        // If a group of nodes are selected, then clicking
+        // one of the nodes in the group shouldn't deselect it,
+        // instead the user must click something else.
+        const foundSelectedNode = (selectedNodes || new Map()).has(
```

Comment:
Nit: Use optional chaining instead of creating new Map, `selectedNodes?.has(...`

Reply:
Optional chaining will be a little confusing because foundSelectedNode should be boolean, but with optional chaining it'll end up as undefined when it's not found. It won't be a huge problem though since the next if statement is truthy, so undefined will end up as false. It's just a preference of what the developers want to read.